### ISSUE TITLE
Update inputs for simplified package install

### ISSUE THREAD

**Issue by MSP-Greg** [msys2/setup-msys2#165]
I'd like to propose adding two inputs, which I think would make the UX a bit better.

Borrowing two samples from the README:

```yaml
  steps:
  - uses: msys2/setup-msys2@v2
    with:
      update: true
      install: >-
        base-devel
        git
```

```yaml
  strategy:
    matrix:
      include:
        - { sys: mingw64, env: x86_64 }
        - { sys: mingw32, env: i686 }
        - { sys: ucrt64,  env: ucrt-x86_64 }  # Experimental!
        - { sys: clang64, env: clang-x86_64 } # Experimental!
  steps:
    - uses: msys2/setup-msys2@v2
      with:
        msystem: ${{matrix.sys}}
        install: mingw-w64-${{matrix.env}}-toolchain
```

Let's combine them and add another package:

```yaml
  strategy:
    matrix:
      include:
        - { sys: mingw64, env: x86_64 }
        - { sys: mingw32, env: i686 }
        - { sys: ucrt64,  env: ucrt-x86_64 }  # Experimental!
        - { sys: clang64, env: clang-x86_64 } # Experimental!
  steps:
    - uses: msys2/setup-msys2@v2
      with:
        msystem: ${{matrix.sys}}
        install: mingw-w64-${{matrix.env}}-toolchain mingw-w64-${{matrix.env}}-openssl base-devel git
```

Now, rewriting the above and adding `install-msys` and `install-mingw` inputs:

```yaml
  strategy:
    matrix:
      sys: [mingw64, mingw32, ucrt64, clang64]
  steps:
    - uses: msys2/setup-msys2@v2
      with:
        msystem: ${{matrix.sys}}
        install-msys: base-devel git
        install-mingw: openssl toolchain
```

I think the workflow is much simpler with the additional inputs.  It's also easier for non-Windows users to get an idea of what it's doing.  Since the `sys` input defines what the package prefix is, we should have the action handle it, not the user.

Not sure if anyone is interested, or whether `install-msys` and `install-mingw` are good input choices.  If there's interest, I could work on a PR...

**Comment by MSP-Greg** [comment]
@eine

Maybe have another read.

> Moreover, we should not force the users to use matrix.sys, because they might be using that for some other purpose. Therefore, I don't think it's desirable to have install-mingw rely on parsing a matrix variable.

The example in the README uses `sys` for a matrix variable.  Users can use whatever variable names they want.  That has nothing to do with the changes I'm suggesting?

I'm suggesting changing/adding inputs so users can clearly see what msys packages are installed (and often used in scripts, regardless of the compiler chain chosen), and removing the need for the redundant `env` matrix variable used in the sample...

**Comment by MSP-Greg** [comment]
No.  My proposal doesn't have to be a breaking change, from your above message:
```yml
install: > 
  mingw-w64-${{matrix.env}}-toolchain
  mingw-w64-${{matrix.env}}-openssl
  base-devel
  git
```

My proposal, which also removes the need for the `env` matrix variable (it's determined by the actions `msystem` input), would be:
```yml
install-mingw: toolchain openssl
install-msys: base-devel git
```
If you prefer separate lines, that's fine too...

**Comment by eine** [comment]
Renaming `install` to `install-*` is a breaking change. Three fields would be needed if we were to proceed with your syntax while avoiding breaking changes (`install`, `install-msys`, `install-mingw`).

The solution I proposed is equivalent to your examples, by just replacing field names `install-msys` and `install-mingw` with `install` and `pacboy`, respectively. Can you please read previous messages carefully?

**Comment by eine** [comment]
Precisely, so that other people do not misunderstand that you want to add two new fields and preserve install, please update your examples to reflect it.

**Comment by MSP-Greg** [comment]
The issue starts with 'I'd like to propose adding two inputs'

**Comment by eine** [comment]
> I noticed #163. Does pacman have a length limit on the 'packages' string?

Please, stay on topic. You can reply in #163 or create a new issue for discussing that.

> #166 closes this?

It does: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue.

> Not from my perspective.

Please, go ahead an update your proposal (as requested) to clearly explain the capability that you want and why you think it is not solved by #166.
